packages/oauth/src/install-provider.ts

```diff
@@ -245,32 +285,84 @@ export class InstallProvider {
    * The return value is an Array of Promises made up of the resolution of each token refresh attempt.
    */
   private async refreshExpiringTokens(tokensToRefresh: string[]): Promise<OAuthV2TokenRefreshResponse[]> {
-    const client = new WebClient(undefined, this.clientOptions);
-
-    const refreshPromises = tokensToRefresh.map(async (refreshToken) => await client.oauth.v2.access({
-      client_id: this.clientId,
-      client_secret: this.clientSecret,
-      grant_type: 'refresh_token',
-      refresh_token: refreshToken,
-    }).catch((e) => e) as OAuthV2TokenRefreshResponse);
+    const refreshPromises = tokensToRefresh
+      .map(async (refreshToken) => this.noTokenClient.oauth.v2.access({
+        client_id: this.clientId,
+        client_secret: this.clientSecret,
+        grant_type: 'refresh_token',
+        refresh_token: refreshToken,
+      }).then((res) => res as OAuthV2TokenRefreshResponse));
 
     return Promise.all(refreshPromises);
   }
 
+  // ------------------------------------------------------
+  // Handling web browser requests form end-users
+  // ------------------------------------------------------
+
   /**
-   * Returns search params from a URL and ignores protocol / hostname as those
-   * aren't guaranteed to be accurate e.g. in x-forwarded- scenarios
-   */
-  private static extractSearchParams(req: IncomingMessage): URLSearchParams {
-    const { searchParams } = new URL(req.url as string, `https://${req.headers.host}`);
-    return searchParams;
+   * Handles the install path (the default is /slack/install) requests from an app installer.
+  */
+  public async handleInstallPath(
+    _req: IncomingMessage,
+    res: ServerResponse,
+    installOptions?: InstallURLOptions,
+  ): Promise<void> {
+    if (installOptions === undefined && this.installUrlOptions === undefined) {
+      const errorMessage = 'To enable the built-in install path handler, you need to pass InstallURLOptions to InstallProvider. ' +
+      "If you're using @slack/bolt, please upgrade the framework to the latest version.";
+      throw new GenerateInstallUrlError(errorMessage);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const _installOptions: InstallURLOptions = installOptions! || this.installUrlOptions!;
+    this.logger.debug(`Running handleInstallPath() with ${_installOptions}`);
+
+    if (this.stateStore === undefined) {
```

Comment:
This never happens in normal use cases but the type is `StateStore | undefined`. Thus, I added validation, just in case.